LayoutTests/fast/dom/gc-abort-controller.html

```diff
@@ -0,0 +1,61 @@
+<!DOCTYPE html>
+<html>
+<head>
+    <meta charset="utf-8">
+    <title>Test AbortController memory leak with event listeners</title>
+    <script src="../../resources/js-test.js"></script>
+</head>
+<body>
+<script>
+jsTestIsAsync = true;
+
+if (!window.testRunner || !window.internals) {
+    testFailed("Requires testRunner and internals.");
+    finishJSTest();
+}
+
+function runTest() {
+    let weakRefs = [];
+    let controllers = [];
+    
+    // Create 500 AbortControllers with event listeners
+    for (let i = 0; i < 500; i++) {
+        let controller = new AbortController();
+        controller.signal.addEventListener('abort', () => {
+            console.log('Event listener for controller ' + i);
+        });
+        controllers.push(controller);
+        weakRefs.push(new WeakRef(controller.signal));
+    }
+    
+    controllers = null;
+    
+    var cnt = 0;
+    let handle = setInterval(() => {
+        gc();
+        cnt++;
+        
+        // Count how many signals are still alive
+        let aliveCount = 0;
+        for (let weakRef of weakRefs) {
+            if (weakRef.deref()) {
+                aliveCount++;
+            }
+        }
+        
+        if (aliveCount === 0) {
+            clearInterval(handle);
+            testPassed("All 500 AbortSignals were garbage collected.");
+            finishJSTest();
+        } else if (cnt >= 500) {
```

Comment:
This check is also risky. No guarantee the first gc collects any objects. We usually set use a setTimeout for the failure case.

Reply:
Adjusted GC and test check timing with setTimeout. I had been using setInterval to retry GC, but I agree it was unintuitive. Thanks!

Reply:
Applied your suggestion to update the test.
Instead of repeatedly triggering GC, I now create 20 AbortSignal objects and check if some of them get collected after a single GC run.
This accounts for the conservative nature of the GC where not all objects may be collected in one pass.